protos/flyteidl/admin/common.proto

```diff
@@ -300,3 +301,56 @@ message RawOutputDataConfig {
     // e.g. s3://bucket/key or s3://bucket/
     string output_location_prefix = 1;
 }
+
+// Span represents a duration trace of Flyte execution. This can refer to either a category, which labels the span with
+// a description, or a reference, which facilitates hierarchical breakdown of spans to capture Flyte entity
+// relationships (ex. workflow -> node -> task).
+message Span {
+    // start_time defines the instance this span began.
+    google.protobuf.Timestamp start_time = 1;
+
+    // end_time defines the instance this span completed.
+    google.protobuf.Timestamp end_time = 2;
+
+    oneof info {
+        // category labels the span with a brief description.
+        CategoricalSpanInfo category = 3;
+
+        // reference allows nesting spans by providing a hierarchical interface to represent Flyte entity
+        // replationships (ex. workflow -> node -> task).
+        ReferenceSpanInfo reference = 4;
+    }
+}
+
+// CategoricalSpanInfo represents a labelled Span, providing a brief attribution for the duration.
+message CategoricalSpanInfo {
+    // Category defines a collection of span categories.
+    enum Category {
+        UNKNOWN = 0;
+        NODE_TRANSITION = 1;
+        EXECUTION_OVERHEAD = 2;
+        EXECUTION_IDLE = 3;
+        PLUGIN_OVERHEAD = 4;
+        PLUGIN_RUNTIME = 5;
```

Comment:
qq: What's different between `PLUGIN_OVERHEAD` and `PLUGIN_RUNTIME`. 
Does `PLUGIN_OVERHEAD` mean the overhead in the propeller when creating the CR?
PLUGIN_RUNTIME is the actual task overhead?

Reply:
great question. so for example starting a k8s pod. `PLUGIN_OVERHEAD` is the time from when propeller performs the `Create` call to when the internal container starts. this covers things like pod scheduling, container pulls, etc. whereas `PLUGIN_RUNTIME`, in this example, is when the  container is actually running.